frontend/dashboard/pages/PageLayout/DashboardHeader/DashboardHeader.tsx

```diff
@@ -20,20 +20,24 @@ import { useRepoPath } from 'dashboard/hooks/useRepoPath';
 import { usePageHeaderTitle } from 'dashboard/hooks/usePageHeaderTitle';
 import { useSubroute } from '../../../hooks/useSubRoute';
 import type { HeaderMenuItem } from './dashboardHeaderMenuItems';
-import { dashboardHeaderMenuItems } from './dashboardHeaderMenuItems';
+import { dashboardHeaderMenuItems, HeaderMenuItemKey } from './dashboardHeaderMenuItems';
 import { StringUtils } from '@studio/pure-functions';
+import { FeatureFlag, shouldDisplayFeature } from 'app-shared/utils/featureToggleUtils';
 
 export const DashboardHeader = () => {
   const pageHeaderTitle: string = usePageHeaderTitle();
+  const headerMenuItems: HeaderMenuItem[] = getHeaderMenuItems();
+  const isMoreThanOneMenuItem: boolean = headerMenuItems.length > 1;
 
   return (
     <StudioPageHeader>
       <StudioPageHeader.Main>
         <StudioPageHeader.Left title={pageHeaderTitle} showTitle />
         <StudioPageHeader.Center>
-          {dashboardHeaderMenuItems.map((menuItem) => (
-            <TopNavigationMenuItem key={menuItem.name} menuItem={menuItem} />
-          ))}
+          {isMoreThanOneMenuItem &&
+            headerMenuItems.map((menuItem: HeaderMenuItem) => (
+              <TopNavigationMenuItem key={menuItem.name} menuItem={menuItem} />
+            ))}
```

Comment:
Could we use `shouldDisplayFeature(FeatureFlag.OrgLibrary)` directly in the return statement? Then we would not need `isMoreThanOneMenuItem`,  `getHeaderMenuItems` and `filterOutOrgLibraryFromMenuItems`.
```suggestion
          {shouldDisplayFeature(FeatureFlag.OrgLibrary) &&
            dashboardHeaderMenuItems.map((menuItem: HeaderMenuItem) => (
              <TopNavigationMenuItem key={menuItem.name} menuItem={menuItem} />
            ))}
```

Reply:
I was thinking about that, but the reason I added it was in case we were to add more elements in the header, e.g., `resources` and `datamodel` and merge it before the `org-library`. But we can add it back in case it happens 👍